odoo/addons/base/models/ir_http.py

```diff
@@ -458,3 +465,66 @@ def _is_allowed_cookie(cls, cookie_type):
     @api.model
     def _verify_request_recaptcha_token(self, action: str):
         return
+
+    @classmethod
+    def _must_check_identity(cls):
+        """
+        Determine whether the current user session requires identity confirmation.
+        :return: A dictionary describing the re-authentication requirement.
+        Possible keys:
+        - `logout` [bool]: True if a full login is required
+        - `check_identity` [bool]: True if an identity check is required
+        - `mfa` [bool]: True if multi-factor authentication is required
+        - `1fa_method` [str]: previously used auth method, to avoid reuse as second factor
+        """
+        return {}
+
+    @classmethod
+    def _check_identity(cls, credential):
+        """
+        Verify the user's identity using the given credentials.
+        Handles both single and multi-factor authentication flows depending on the
+        current session state and configured timeout rules.
+
+        :param dict credential: A dictionary containing authentication data. Must include
+            a "type" key (e.g., "password", "totp", "webauthn"). If empty, the method
+            returns the list of available authentication methods.
+
+        :return: A dictionary indicating the outcome of the identity check:
+
+            - {"auth_methods": [...]} if no credential is provided,
+            - {"mfa": True, "auth_methods": [...]} if a second factor is required,
+            - None if re-authentication is complete.
+
+        :rtype: dict or None
+        """
+        user = request.env.user
+        auth_methods = user._get_auth_methods()
+
+        reauth_requirements = cls._must_check_identity()
+        assert set(reauth_requirements).issubset(('logout', 'check_identity', 'mfa', '1fa_method'))
```

Comment:
```suggestion
        assert set(reauth_requirements).issubset(('logout', 'check_identity', 'mfa', '1fa_method')), reauth_requirements
```
so the AssertionError logs the reauth_requirements